### DISCUSSION TITLE
(09) TravelPlan [P.20220608]

### DISCUSSION THREAD

**Discussion by progettistamatto** [sapienzastudentsnetwork/basi-di-dati-2#30] [Progetti]
In questa discussion a tema il progetto [TravelPlan](https://elearning.uniroma1.it/pluginfile.php/1350504/mod_folder/content/0/%285%29%20P%20-%20Progetti/%2809%29%20TravelPlan%20-%20BD2.pdf), discuteremo di eventuali proposte di soluzione, dubbi o scelte progettuali in merito alla realizzazione del progetto in questione

**N.B.:** il progetto P.20220608 si contraddistingue, rispetto al testo oggetto dell'omonimo esame del 7 novembre 2019, per un'enfasi nelle specifiche del progetto sul fatto che il sistema debba permettere tra le cose agli utenti registrati di partecipare a viaggi:
-  > Il sistema deve permettere agli utenti registrati **di partecipare a viaggi** e di creare viaggi in qualità di organizzatori, aggiungendo a essi attività di vario tipo.

E per un use case richiesto in più:
-  > I responsabili del sistema devono poter calcolare, data una città di destinazione, per ognuno dei 12 mesi dell’anno, il numero di viaggi organizzati in quel mese nell’ultimo anno solare.

**Comment by progettistamatto** [comment]
Per questo progetto è disponibile [[QUI]](https://github.com/Exyss/university-notes/raw/main/Bachelor/Secondo%20Anno/Basi%20di%20Dati%20II/TravelPlan.pdf) una soluzione proposta da Simone Bianco (Exyss)

> [!Warning]
_Le soluzioni di Simone Bianco fanno uso degli **Entity-Relationship (ER)** Diagrams su cui il corso si basava fino all'Anno Accademico 2023/2024, mentre **attualmente vengono utilizzati gli UML Class** Diagram. Sebbene i due siano estremamente simili, vi sono **alcune discordanze tra di loro**, in particolare l'**inversione di posizione dei vincoli di molteplicità agli estremi delle relationship**. Nonostante ciò, il rimanente delle fasi di analisi e progettazione delle soluzioni proposte risulta inaffetto da tale cambiamento. Per tanto, una volta comprese le differenze tra ER e Classi UML, le soluzioni risultano normalmente fruibili._

Rispondi a questo commento taggando @Exyss per eventuali dubbi o segnalazioni in merito

**Reply by Exyss** [reply]
Update: ho modificato la soluzione affinché abbia anche lo use case aggiuntivo al suo interno, dunque ora é una soluzione valida sia per il progetto che per l'esame

**Comment by progettistamatto** [comment]
Per questo progetto è disponibile [[QUI]](https://github.com/SimoneLid/BD2/raw/main/Progetti/TravelPlan/TravelPlan.pdf) una soluzione proposta da Simone Lidonnici (@SimoneLid)

**Comment by daromacarlo** [comment]
https://github.com/daromacarlo/Triennale-in-Informatica/tree/main/Basi%20di%20dati/Basi%20di%20dati%20MODULO%202/Progetti/9.TravelPlan

**Comment by CosaFaSelect1** [comment]
Per questo progetto è disponibile [[QUI]](https://github.com/CasuFrost/university-notes-Informatica/raw/main/Secondo%20Anno/Secondo%20Semestre/Basi%20di%20Dati%202/Progetti/TravelPlan/TravelPlan.pdf) una soluzione proposta da Marco Casu
